tdms/src/interpolate.cpp

```diff
@@ -1393,3 +1272,177 @@ void interpolateTimeDomainFieldCentralE_TM(  double ***Exy, double ***Exz, doubl
   *Ey = 0.;
   *Ez = Ezy[k][j][i]+Ezx[k][j][i];
 }
+
+/**
+ * @brief Determines the appropriate interpolation scheme to use
+ * 
+ * @param[in] cells_in_direction The number of Yee cells in the interpolation direction of interest
+ * @param[in] cell_id The current ID (in this dimension) of the Yee cell
+ * @return int Equal to BAND_LIMITED, INTERP1, INTERP2, INTERP3, indicating the appropriate interpolation scheme
+ */
+int determineInterpScheme(int cells_in_direction, int cell_id) {
+  // interpolation is impossible if the total number of cells in this direction is <4
+  if (cells_in_direction <= 4) { throw out_of_range("Error: computational domain has fewer than 4 cells in at least 1 dimension, cubic and band-limited interpolation impossible.\n"); }
+  // Yee cell with index <=0 doesn't exist (indexing starts from 1)
+  else if (cell_id <= 0) { throw out_of_range("Error: Interior Yee cell index <=0 requested (must be >=1).\n");}
+  // Yee cell with index >=cells_in_direction doesn't exist
+  else if (cell_id > cells_in_direction) { throw out_of_range("Error: requested Yee cell index beyond maximum number of Yee cells.\n");}
```

Comment:
```suggestion
  else if (cell_id >= cells_in_direction) { throw out_of_range("Error: requested Yee cell index beyond maximum number of Yee cells.\n");}
```

Reply:
maybe the comment is wrong but the code was correct? if it's 1 indexed. sorry for the perhaps poor suggestion!

Reply:
No I think your original comment is correct - the documentation says that Yee cell IDs start from 1,1,1, BUT C indexes start at 0,0,0. It's also compounded by the fact that we _can_ interpolate when we hit the last cell, but can't when we hit the first, since Yee cells are associated to field values at gridpoints _after_ their origin. 
So `cell_id = 0` errors since we would need the field value in the "-1"th Yee cell, whilst `cell_id = cells_in_direction` errors due to memory access faults.
`cell_id = cells_in_direction-1` _doesn't_ error since we can use `interp3`, as there is a field value after this point.